### ISSUE TITLE
roberta performance degradation of 1.6 release

### ISSUE THREAD

**Issue by world2vec** [pytorch/xla#2460]
Hi,
I see the performance degradation of almost 2 times in colab to fine tuning roberta-base from transformers==2.11.0 compared to torch 1.5 with version 20200516. This is abnormal, any clue?

torch1.5 with xla version 20200516:
Train Epoch:1, lr:0.00003000, loss:loss:0.26035690: 100% 148/148 [04:19<00:00,  1.75s/it]
Val Epoch:1, loss:val_loss:0.26267286: 100% 38/38 [00:20<00:00,  1.89it/s]

torch1.6 with xla 1.6:
Train Epoch:0, lr:0.00003000, loss:loss:0.58976904: 100% 148/148 [17:36<00:00,  7.14s/it]
Val Epoch:0, loss:val_loss:0.27056275: 100% 38/38 [01:44<00:00,  2.74s/it]
Train Epoch:1, lr:0.00003000, loss:loss:0.26036847: 100% 148/148 [06:29<00:00,  2.63s/it]
Val Epoch:1, loss:val_loss:0.26261912: 100% 38/38 [00:14<00:00,  2.67it/s]

Another question is why the first epoch is always very slow?

Thanks

**Comment by jysohn23** [comment]
Which finetuning task are you talking about? Can you post a repro Colab?

Also, first epoch is typically slow because the TPU needs to compile the binary for the graph it's seeing for the first time. But in subsequent epochs, since the graphs are static, they hit the binary cache, and can just re-use the compiled binaries on TPU.

**Comment by world2vec** [comment]
I have rollback to 1.5.
I can not public my code directly. I am busy now, will have a detail look and give a notebook to reproduce it when I am free. 
Thanks

**Comment by byronyi** [comment]
> Which finetuning task are you talking about? Can you post a repro Colab?
> 
> Also, first epoch is typically slow because the TPU needs to compile the binary for the graph it's seeing for the first time. But in subsequent epochs, since the graphs are static, they hit the binary cache, and can just re-use the compiled binaries on TPU.

Any chance subsequent training jobs could re-use the same cache (e.g. for debugging, if the model doesn't change)? I saw some compilation cache service work recently in the TF side, wondering if it could be used for torch-xla.

**Comment by jysohn23** [comment]
Hi @byronyi I believe the commits you saw were for open sourcing some of the caches on the `tensorflow/tensorflow` repo. This cache has actually been already enabled on the TPU side regardless of TF/PyTorch. So as long as you run two training runs that have the exact same graphs, the second run should hit the TPU compilation cache unless you restarted the TPU's state between the first and second runs, which would empty the cache. You can check TPU side metrics in the metrics report using our metrics report under `Xrt` prefixed metrics (`XrtExecute`, `XrtCompile`, etc).

**Comment by world2vec** [comment]
Still busy. 
It's a sequence labeling problem, a BiLSTM layer added.
I guess I will be free after December.
Thanks.

**Comment by world2vec** [comment]
Will work on December

**Comment by world2vec** [comment]
Did not reproduce it. see notebook [1.5](https://drive.google.com/file/d/1U2rWqplq0yuytQgiL7TtojjgAWM576ym/view?usp=sharing)
and [1.6](https://drive.google.com/file/d/1V2VvXUPlqsuXl9N18N2A3Ns7aOJNaIsr/view?usp=sharing), they have similar performance.
Thanks.
